packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -998,6 +1003,13 @@ abstract class PopupRoute<T> extends ModalRoute<T> {
 ///     // Do something
 ///   }
 ///
+///   @override
+///   Widget build(BuildContext context) => new Container();
+///
+/// }
+///
+/// class RouteAwareWidget extends StatefulWidget {
+///   State<RouteAwareWidget> createState() => new RouteAwareWidgetState();
 /// }
 /// ```
 class RouteObserver<T extends Route<dynamic>> extends NavigatorObserver {
```

Comment:
For the Firebase Analytics use case we'd like to be informed about `PageRoute`s that are pushed onto other `PageRoute`s (or popped from them) to track page transitions (see https://github.com/flutter/plugins/blob/master/packages/firebase_analytics/lib/observer.dart#L51). This `RouteObserver` enables us to do just that.

During the initial PR we saw that a PageRouteObserver could be generalized to observe any type of route changes and what you see here is the result of that. If you want to be informed about route changes of any type, you use a `RouteObserver<Route>`. If you are only interested PageRoute changes, you use `RouteObserver<PageRoute>`.